### PR TITLE
Expanded Correlation Selection Options.

### PR DESCRIPTION
**Pull request by aewallwi**
Options to select cross-correlations and autocorrelations in power spectrum with new args `include_autocorrs` and `exclude_crosscorrs` in `pspec_run`. This allows us to create autocorrelation power-spectra as a diagnostic. Depends on #314

### CODE REVIEW

File: hera_pspec/grouping.py [hunk 1]
```diff
@@ -1103,10 +1103,10 @@ def bootstrap_average_blpairs(uvp_list, blpair_groups, time_avg=False,
                         for blpg in blpair_groups]
         blpair_groups = new_blp_grps
 
-    # Homogenise input UVPSpec objects in terms of available polarizations 
+    # Homogenise input UVPSpec objects in terms of available polarizations
     # and spectral windows
     if len(uvp_list) > 1:
-        uvp_list = uvpspec_utils.select_common(uvp_list, spws=True, 
+        uvp_list = uvpspec_utils.select_common(uvp_list, spws=True,
```
Comment (aewallwi) [inline hera_pspec/grouping.py:1109]: This miss was caused by white space removal.

File: scripts/generate_pstokes_run.py [hunk 2]
```diff
@@ -24,4 +24,4 @@
         uvd_output += pstokes.construct_pstokes(uvd, uvd, pstokes=p)
 
 # write file.
-uvd_output.write_uvh5(args.outputdata, clobber=args.clobber)
+uvd_output.write_uvh5(args.outputdata, clobber=args.clobber)
```
Comment (jsdillon) [inline scripts/generate_pstokes_run.py:27]: Why did you remove the newline?

File: hera_pspec/utils.py [hunk 3]
```diff
@@ -1342,7 +1372,7 @@ def uvp_noise_error(uvp, auto_Tsys=None, err_type='P_N', precomp_P_N=None, P_SN_
     little_h : bool, optional
         Use little_h units in power spectrum.
         Default is True.
-        
+
```
Comment (jsdillon) [inline hera_pspec/utils.py:1375]: This whitespace doesn't seem standard.

**Review by jsdillon — CHANGES_REQUESTED** [review]
Seems basically fine. Just a handful of minor issues, mostly with documentation.

File: hera_pspec/grouping.py [hunk 4]
```diff
@@ -1175,6 +1175,7 @@ def bootstrap_average_blpairs(uvp_list, blpair_groups, time_avg=False,
     # using the bootstrap-sampled blpair weights
     uvp_avg = []
     for i, uvp in enumerate(uvp_list):
+        # using the bootstrap-sampled blpair weights
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_pspec/grouping.py:1178]: This must be a merge artifact. Thanks for catching! I've deleted this.

File: hera_pspec/pspecdata.py [hunk 5]
```diff
@@ -3963,6 +3963,17 @@ def pspec_run(dsets, filename, dsets_std=None, cals=None, cal_flag=True,
             - `filter_factors`: list of floats (or float) specifying how much
                                 power within each filter window is to be
                                 suppressed.
+    include_autocorrs : bool, optional
+        If True, include power spectra of autocorrelation visibilities.
+        Default is False.
+
+    xant_flag_thresh : float, optional
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_pspec/pspecdata.py:3970]: Maybe I selected the wrong rebase option? Fixed!

File: scripts/generate_pstokes_run.py (hunk 2 above)
Comment (aewallwi) [reply scripts/generate_pstokes_run.py:27]: Not sure. I've put it back in.

File: hera_pspec/utils.py (hunk 3 above)
Comment (aewallwi) [reply hera_pspec/utils.py:1375]: I think this has been standardized now.

**Comment by aewallwi** [comment]
Thanks for the timely review @jsdillon. I think this is ready for another look when you have a chance!

**Review by jsdillon — APPROVED** [review]

**Pull request merged**